state/dbstore_test.go

```diff
@@ -77,9 +79,18 @@ func TestDBStore(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer persistedStore.Close()
 
 	testPersistedStore(t, persistedStore)
+
+	persistedStore.Close()
```

Comment:
Without the defer, `persistedStore.Close()` won't run if `testPersistedStore` fails with a `t.Fatal`.
But because you are using the same `dir` to create the test store, probably the only way to address this is to put a `defer ...Close()` inside `testPersistedStore`?

Reply:
Nice catch! Fixed.